applicationconfigurations/workflow.md

```diff
@@ -0,0 +1,219 @@
+# An Example Workflow
+
+This section is non-normative. OAM compliant tooling need not implement this section.
+
+This section uses a fictional tool called `oamctl` to illustrate a workflow pattern for OAM app operations.
+
+### Deploying two components with parameter overrides
+The following example shows two separate components:
+ - a front-end web app in a container, run as a core container workload.
+ - a back-end Cassandra database, represented by an extended workload.
+
+```yaml
+apiVersion: core.oam.dev/v1alpha2
+kind: Component
+metadata:
+  name: frontend
+  annotations:
+    version: v1.0.0
+    description: "A simple webserver"
+spec:
+  workload:
+    apiVersion: core.oam.dev/v1alpha2
+    kind: containerWorkload
+    metadata:
+      name: sample-workload
+    spec:
+      osType: linux
+      containers:
+      - name: web
+        image:
+          name: example/charybdis-single:latest
+        resources:
+          cpu:
+            required: 1.0
+          memory:
+            required: 100MB
+        env:
+        - name: MESSAGE
+          value: default
+  parameters:
+  - name: message
+    description: The message to display in the web app.  
+    required: true
+    type: string
+    value: "Hello from my app, too"
+    fieldPaths:
+    - "spec.containers[0].env[0].value"
+```
+
+```yaml
+apiVersion: core.oam.dev/v1alpha1
+kind: Component
+metadata:
+  name: backend
+  annotations:
+    version: v1.0.0
+    description: "Cassandra database"
+spec:
+  workload:
+    apiVersion: data.oam.dev/v1
+    kind: Cassandra
+    spec:
+      maxStalenessPrefix: 100000
+      defaultConsistencyLevel: Eventual
+  parameters:
+    - name: maxStalenessPrefix
+      description: Max stale requests.
+      type: int
+      value: 100000
+      fieldPaths:
+      - "spec.maxStalenessPrefix"
+    - name: defaultConsistencyLevel
+      description: The default consistency level
+      type: string
+      value: "Eventual"
+      fieldPaths:
+      - "spec.defaultConsistencyLevel"
+```
+
+Note that each component allows certain parameters to be overridden. For example, the `message` parameter is exposed for configuration in the frontend component. Within the component config, the parameter is piped to an environment variable where the component code can read the value.
+
+Components can be designed for reuse by expose such parameters, as these parameters can be tweaked for different deployment scenarios.
```

Comment:
Suggestion 

"Components are designed for reuse by exposing  parameters to be configured in different deployment scenarios."

Reply:
good